server/apps/distributed-app/docker-compose-with-pulsar.yml

```
@@ -34,10 +21,11 @@ services:
       - "8000:8000"
     volumes:
       - ./sample-configuration/pulsar.properties:/root/conf/pulsar.properties
-      - ./sample-configuration/keystore:/root/conf/keystore
+      - ./sample-configuration/queue.properties:/root/conf/queue.properties
```

Did you retest the Pulsar docker-compose with this change?

The current sample `queue.properties` file comes with `queue.choice=RABBITMQ`, therefore I really doubt if the Pulsar mail queue is actually being used here.

```suggestion
      - ./sample-configuration/queue-pulsar.properties:/root/conf/queue.properties
```

---

Yes, I already tested it. Because of mounting queue.properties, I have to add @Inject to RabbitMQConfiguration and ReactorRabbitMQChannelPool constructor

---

" Because of mounting queue.properties, I have to add @Inject to RabbitMQConfiguration and ReactorRabbitMQChannelPool constructor"

Can you be more elaborate?

---

Because there is no injectable constructor for RabbitMQConfiguration and ReactorRabbitMQChannelPool error when I build docker-compose

The error is sth like this:
11) [Guice/MissingConstructor]: No injectable constructor for type ReactorRabbitMQChannelPool$Configuration.
class ReactorRabbitMQChannelPool$Configuration does not have a @Inject annotated constructor or a no-arg constructor.

---

That is not the rationale. 
The above error is just the surface of the ice IMO.
We did not need the @Inject before, and which change forced us to change that? Can you investigate more to see if we can fix it another way?

FYI, before we already declared a bean (on top of a method) for `ReactorRabbitMQChannelPool`: https://github.com/apache/james-project/blob/master/server/container/guice/queue/rabbitmq/src/main/java/org/apache/james/modules/queue/rabbitmq/RabbitMQModule.java#L134. That is why we do not need to declare a bean using @Inject (on top of the constructor).

I would suspect the James with Pulsar server is missing that bean (on top of a method) module...

---

Ok, let me check more

---

Yep, you're right, I remove these @Inject and rebuild the project and everything still works. This error should come from other reasons.

---

Can we have a sample `queue-pulsar.properties` for Pulsar docker compose?